functions.sh

```diff
@@ -55,7 +55,11 @@ sonic_get_version() {
     local latest_tag=$(git describe --tags --abbrev=0)
     local branch_name=$(git rev-parse --abbrev-ref HEAD)
     if [ -n "$(git status --untracked-files=no -s --ignore-submodules)" ]; then
-        local dirty="-dirty-$DIRTY_SUFFIX"
+        if [ -z "$DIRTY_SUFFIX" ]; then
+            local dirty="-dirty-"$(date +%Y%m%d\.%H%M%S)
```

Comment:
DIRTY_SUFFIX is defined in slave.mk, why do we need this check?

Reply:
I tried changes like moving DIRTY_SUFFIX before SONIC_GET_VERSION
```
DIRTY_SUFFIX="$(shell date +%Y%m%d\.%H%M%S)"
SONIC_GET_VERSION=$(shell . functions.sh && sonic_get_version)
```
And even export DIRTY_SUFFIX before docker build.   They didn't work,  tag like "warm-reboot.0-dirty-" was populated.
I guess there is some evaluation issue with layered macro with shell command in Makefile.  

If you have any new idea or different way of handling it,  I may give it a try.

Reply:
does DIRTY_SUFFIX work now?

Reply:
DIRTY_SUFFIX works if sonic_get_version() is called from another shell script. But calling it in Makefile is not working.  I have not figured out how to do it.

Since DIRTY_SUFFIX is only used in sonic_get_version(), but defined in slave.mk,  maybe it should be moved to sonic_get_version() to get rid of the issue introduced with jumping from Makefile to shell and back to Makefile again.